### PR TITLE
CDAP-1206 columns in a partitioned table were not showing up

### PR DESCRIPTION
**Pull request by albertshau**
in our table info call. To fix this, we have to make a separate
call to the hive metastore to get the table schema. In addition,
adding a table property to all tables created by CDAP so that
we can tell which hive tables are cdap managed and which are not.

### CODE REVIEW

**Comment by albertshau** [comment]
tested on standalone, hdp, and cdh

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java [hunk 1]
```diff
@@ -481,18 +481,41 @@ public TableInfo getTableInfo(@Nullable String database, String table)
...
+      ImmutableList.Builder<TableInfo.ColumnInfo> partitionKeysBuilder = ImmutableList.builder();
       for (FieldSchema column : tableInfo.getPartitionKeys()) {
-        partitionKeysBuilder.add(new TableInfo.ColumnInfo(column.getName(), column.getType(),
-                                                                     column.getComment()));
+        TableInfo.ColumnInfo columnInfo = new TableInfo.ColumnInfo(column.getName(), column.getType(),
+                                                                   column.getComment());
+        partitionKeysBuilder.add(columnInfo);
+        // add partition keys to the schema as well, since they show up when you do a 'describe <table>' command.
+        schemaBuilder.add(columnInfo);
+      }
```
Comment (anew) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: I wonder what happens if a column of the schema is used as a partitioning key. I assume you will see that column twice here? Perhaps you should filter such duplicates?

**Comment by anew** [comment]
Only one comment. Perhaps you can test that case before you merge?

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java (hunk 1 above)
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: will do

**Comment by albertshau** [comment]
tried it out and hive will not let you create a table where a column is both a field and a partition key. Though you never know what will be allowed or what will get returned in the future so will protect against duplication anyway.

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT586

**Comment by anew** [comment]
Thanks for testing that, Albert.

**Comment by anew** [comment]
LGTM

**Pull request merged**